src/two_finger_grasp_generator.cpp

```diff
@@ -0,0 +1,917 @@
+    // {
+    //   intersection[0]= u;
+    //   intersection[1]= -width / 2.0;
+    //   intersection[2]= v;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::LIME_GREEN, 0.005);
+    // }
+    return true;
+  }
+
+  // check if line segment intersects YZ faces of cuboid (x = +/- depth/2)
+  t = (depth / 2.0 - point_a[0]) / (point_b[0] - point_a[0]);
+  if (intersectionHelper(t, point_a[1], point_a[2], point_b[1], point_b[2], width, height, u, v))
+  {
+    // if (verbose_)
+    // {
+    //   intersection[0]= depth / 2.0;
+    //   intersection[1]= u;
+    //   intersection[2]= v;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::RED, 0.005);
+    // }
+    return true;
+  }
+
+  t = (-depth / 2.0 - point_a[0]) / (point_b[0] - point_a[0]);
+  if (intersectionHelper(t, point_a[1], point_a[2], point_b[1], point_b[2], width, height, u, v))
+  {
+    // if (verbose_)
+    // {
+    //   intersection[0]= -depth / 2.0;
+    //   intersection[1]= u;
+    //   intersection[2]= v;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::PINK, 0.005);
+    // }
+    return true;
+  }
+
+  // no intersection found
+  return false;
+}
+
+bool TwoFingerGraspGenerator::intersectionHelper(double t, double u1, double v1, double u2, double v2, double a,
+                                                 double b, double& u, double& v)
+{
+  // plane must cross through our line segment
+  if (t >= 0 && t <= 1)
+  {
+    u = u1 + t * (u2 - u1);
+    v = v1 + t * (v2 - v1);
+
+    if (u >= -a / 2 && u <= a / 2 && v >= -b / 2 && v <= b / 2)
+      return true;
+  }
+
+  return false;
+}
+
+double TwoFingerGraspGenerator::scoreFingerGrasp(const Eigen::Isometry3d& grasp_pose_tcp,
+                                                 const TwoFingerGraspDataPtr& grasp_data,
+                                                 const Eigen::Isometry3d& object_pose, double percent_open)
+{
+  ROS_DEBUG_STREAM_NAMED("grasp_generator.scoreGrasp", "starting to score grasp...");
+
+  // get portion of score based on the gripper's opening width on approach
+  double width_score = TwoFingerGraspScorer::scoreGraspWidth(grasp_data, percent_open);
+
+  // get portion of score based on the pinchers being down
+  Eigen::Vector3d orientation_scores =
+      TwoFingerGraspScorer::scoreRotationsFromDesired(grasp_pose_tcp, ideal_grasp_pose_);
+
+  // get portion of score based on the distance of the grasp pose to the object pose
+
+  // NOTE: when this function is called we've lost the references to the acutal size of the object.
+  // max_distance should be the length of the fingers minus some minimum amount that the fingers need to grip an object
+  // since we don't know the distance from the centoid of the object to the edge of the object, this is set as an
+  // arbitrary number given our target object set
+  double distance_score = TwoFingerGraspScorer::scoreDistanceToPalm(grasp_pose_tcp, grasp_data, object_pose,
```

Comment:
Why have we lost references to the size, why not allow the user to pass it in as a parameter?

Reply:
legacy code. I think the object size should probably be put in the candidate along with it's pose but I didn't want to touch the finger code because it was hard to follow.

Reply:
fair